lang/java/reef-common/src/main/java/org/apache/reef/driver/catalog/ResourceCatalog.java

```diff
@@ -56,6 +56,13 @@
    */
   NodeDescriptor getNode(String nodeId);
 
+  /**
+   * We are not using this. It will be removed in the future. In order to do
+   * evaluator requests with specific rack names or node names, you should take
+   * a look at {@link EvaluatorRequest} new API.
+   *
+   */
+  @Deprecated
   public interface Descriptor {
 
     String getName();
```

Comment:
we are actually using this.. I believe we should create another interface, something like Nameable, in org.apache.io.naming.... Having an inner interface like this doesn't make much sense. It was mainly use for asking for instanceof...